### ISSUE TITLE
Is it possible to pass `fig`ure to `FacetGrid`?

### ISSUE THREAD

**Issue by tomchor** [mwaskom/seaborn#3520]
I frequently use [`xmovie`](https://github.com/jbusecke/xmovie) to plot animations from [`xarray`](https://docs.xarray.dev/en/stable/) objects since it allows for parallelization of movie creation and provides a really simple and interface for it. The problem is that `xmovie` uses one `matplotlib.pyplot.figure` instance that gets written to over and over (to save time, instead of creating a different figure for each frame).

Because of that, we can't use `xmovies` + `FaceGrids` (which is sometimes used by `xarray`), since it seems like every time a `FacetGrid` is created, a new `matplotlib.pyplot.figure` is also created.

Is this true or am I misunderstanding the situation? If it's true, can we add the possibility of passing a figure object to `FacetGrid`?

Thanks

**Comment by mwaskom** [comment]
You are correct that this is not currently possible. If you search, you will find some previous discussion of it. Also, I think that `xarray` ships their own version of `FacetGrid`, which is inspired by (by technically independent from) seaborn's?
